x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,379 @@
+          fieldStatsControlsApi.dataViewId$
+            .pipe(
+              skip(1),
+              skipWhile((dataViewId) => !dataViewId && !defaultDataViewId),
+              switchMap((dataViewId) => deps.data.dataViews.get(dataViewId ?? defaultDataViewId))
+            )
+            .subscribe((nextSelectedDataView) => {
+              dataViews$.next([nextSelectedDataView]);
+            })
+        );
+      }
+
+      const api = buildApi(
+        {
+          ...timeRangeApi,
+          ...titlesApi,
+          ...fieldStatsControlsApi,
+          ...dataLoadingApi,
+          getTypeDisplayName: () =>
+            i18n.translate('xpack.dataVisualizer.fieldStats.typeDisplayName', {
+              defaultMessage: 'field statistics',
+            }),
+          isEditingEnabled: () => true,
+          onEdit: async () => {
+            try {
+              const { resolveEmbeddableFieldStatsUserInput } = await import(
+                './resolve_field_stats_embeddable_input'
+              );
+              const chartState = serializeFieldStatsChartState();
+              const nextUpdate = await resolveEmbeddableFieldStatsUserInput(
+                coreStart,
+                pluginStart,
+                parentApi,
+                uuid,
+                false,
+                chartState,
+                fieldStatsControlsApi
+              );
+
+              fieldStatsControlsApi.updateUserInput(nextUpdate);
+            } catch (e) {
+              return Promise.reject(e);
+            }
+          },
+          dataViews: dataViews$,
+          serializeState: () => {
+            const dataViewId = fieldStatsControlsApi.dataViewId$?.getValue();
+            const references: Reference[] = dataViewId
+              ? [
+                  {
+                    type: DATA_VIEW_SAVED_OBJECT_TYPE,
+                    name: FIELD_STATS_DATA_VIEW_REF_NAME,
+                    id: dataViewId,
+                  },
+                ]
+              : [];
+            return {
+              rawState: {
+                ...serializeTitles(),
+                ...serializeTimeRange(),
+                ...serializeFieldStatsChartState(),
+              },
+              references,
+            };
+          },
+        },
+        {
+          ...timeRangeComparators,
+          ...titleComparators,
+          ...fieldStatsControlsComparators,
+        }
+      );
+
+      const reload$ = fetch$(api).pipe(
+        skipWhile((fetchContext) => !fetchContext.isReload),
+        map(() => Date.now())
+      );
+      const reset$ = resetData$.pipe(skip(1), distinctUntilChanged());
+
+      const onTableUpdate = (changes: Partial<DataVisualizerTableState>) => {
+        if (isDefined(changes?.showDistributions)) {
+          fieldStatsControlsApi.showDistributions$.next(changes.showDistributions);
+        }
+      };
+
+      const addFilters = (filters: Filter[], actionId: string = ACTION_GLOBAL_APPLY_FILTER) => {
```

Comment:
Blocking errors are for when embeddable is broken in an unrecoverable way. If adding filters fails, the embeddable can still function. Instead of setting a blocking error, how about displaying a warning toast when addFilters fails?

Reply:
Thanks for the note. It would be good to document somewhere that 1) Embeddables should not throw error and instead use blockingError , and 2) best practices for using blockingError so other teams can implement embeddable consistently. Possible I missed it, but this previous pattern is used in multiple places in our current code base and probably should be fixed.